tests/testsuite/update.rs

```diff
@@ -1659,3 +1659,99 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_compatible() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                compatible = "1.0"
+                incompatible = "1.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "2.0.0").publish();
+
+    p.cargo("update")
+        .with_stderr(
+            "\
+[UPDATING] `[..]` index
+[UPDATING] compatible v1.0.0 -> v1.0.1
+",
+        )
+        .run();
+}
+
+#[cargo_test]
+fn update_incompatible() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"[package]
+name = "bar"
+version = "0.0.1"
+edition = "2015"
+authors = []
+
+[dependencies]
+# Check if formatting is preserved
+compatible  =  "1.0"  # This line is unchanged
+incompatible  =  "1.0"  # This line gets partially rewritten
+"#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("check").run();
+
+    Package::new("compatible", "1.0.1").publish();
+    Package::new("incompatible", "2.0.0").publish();
+
+    p.cargo("update --incompatible")
+        // .env("CARGO_LOG", "cargo::ops=trace,cargo::util::toml=debug")
+        .with_stderr(
+            "\
+[UPDATING] `[..]` index
+[UPGRADING] incompatible ^1.0 -> v2.0.0
```

Comment:
Should we use https://github.com/killercup/cargo-edit/blob/522160cdfd057bfc993034ce8d5f776782acf609/src/version.rs#L145

Reply:
Done.